examples/src/DemoApplication.mjs

```diff
@@ -302,28 +298,44 @@ export default class DemoApplication extends PIXI.Application
         // https://github.com/orgs/pixijs/projects/2/views/4?pane=issue&itemId=48582986
         const toggleFilter = (enabled) =>
         {
-            if (enabled)
+            if (options.fishOnly)
             {
-                if (options.fishOnly)
+                const fishFilters = [...this.fishFilters];
+
+                if (enabled)
                 {
-                    this.fishFilters.push(filter);
+                    fishFilters.push(filter);
                 }
                 else
                 {
-                    this.pondFilters.push(filter);
-                }
-            }
-            else if (options.fishOnly)
-            {
-                const index = this.fishFilters.indexOf(filter);
+                    const index = fishFilters.indexOf(filter);
 
-                if (index !== -1) this.fishFilters.splice(index, 1);
+                    if (index !== -1) fishFilters.splice(index, 1);
+                }
+                this.fishFilters = fishFilters;
+                this.fishes.forEach((fish) => { 
+                    fish.filters = fishFilters;
+                });
             }
             else
             {
-                const index = this.pondFilters.indexOf(filter);
+                const pondFilters = [...this.pondFilters];
+
+                if (enabled)
+                {
+                    pondFilters.push(filter);
+                }
+                else
+                {
+                    const index = pondFilters.indexOf(filter);
+
+                    if (index !== -1) pondFilters.splice(index, 1);
+                }
 
-                if (index !== -1) this.pondFilters.splice(index, 1);
+                this.pondFilters = pondFilters;
+                // TODO: seems like a bug, requiring invalidation
+                this.pond.filters = [];
```

Comment:
@GoodBoyDigital any idea why we'd need to invalidate the filters before applying them? This doesn't work without this line.